db/1.9/modules/ROOT/pages/database-connector-query-single.adoc

```diff
@@ -0,0 +1,204 @@
+. In the *Mime Type* tab, set the *Mime Type* field to `application/json`.
+. In the *Advanced* tab, set the *Allowed methods* field to `GET`.
+. Drag a *Set Variable* component to the right of the *Listener* source. +
+This component creates a new variable to save the database results that will be obtained from the *Single Query* operation.
+. Set the *Name* field to `bookCollection` and the *Value* field to `#[[]]`.
+
+.Set Variable configuration
+image::database-querysingle-example-3.png[Set Variable configuration]
+
+=== Configure the For Each Component, the Single Query Operation and Set Payload Component
+
+Continue creating the Mule application using a *For Each* component that iterates over the number of books requested in the *HTTP Listener*, retrieves the books information using the *Single Query* operation, save the results in a payload variable:
+
+. Drag a *For Each* component to the right of *Set Variable*. +
+. Set the *Collection* field to `#[1 to attributes.uriParams.maxId]`. +
+This expression iterates the collection of books in the table, from the first book (`1`) to the number of books set in the `maxId` parameter when performing the HTTP request `/select/book/{maxId}`.
++
+.For Each configuration
+image::database-querysingle-example-4.png[For Each configuration]
++
+[start=3]
+. Drag the *Single query* operation inside the *For Each* component.
+. Set the *Connector configuration* field to `Database_Config` to connect to the MySQL database configuration.
+. Set the *SQL Query Text* field to `SELECT id, title, author FROM BOOK WHERE id = :id`. +
+This query selects the books information from the database.
+. Set the *Input Parameters* field to `![CDATA[#[{'id': payload }]]]`. +
+This expression maps the key parameter `id` (referenced in the previous *SQL Query Text* expression) to `payload` which is the result value of books information retrieved using the `SELECT` query.
++
+.Single query configuration
+image::database-querysingle-example-5.png[Single query configuration]
++
+[start=7]
+. Drag another *Set Variable* component to the right of the *Single query* operation.
+. Set the *Name* field to `bookCollection` and the *Value* field to `#[vars.bookCollection ++ [payload]]`. +
+The original variable `bookCollection` now saves the `payload` of the retrieved book information.
+. Drag a *Set Payload* component to the right of the *For Each* component.
+. Set the *Value* field to `#[vars.bookCollection]` to save the variable content as a new output payload.
+
+=== Run and Test the Mule Application
+
+To complete and test the Mule application:
+
+. Save the project in Studio.
+. Test the app by using the following curl command in your terminal: `curl localhost:8081/select/book/7`. +
+This command selects the first 7 books of the table, regardless of the maximum number of connections (`5`) that is configured in the pool.
```

Comment:
7 > seven
"regardless of the maximum number of connections (`5`) configured in the pool."